lib/Marlin/Marlin/src/feature/prusa/MMU2/mmu2.cpp

```diff
@@ -346,13 +346,12 @@ bool MMU2::rx_str_P(const char *str) {
     uint8_t i = strlen(rx_buffer);
 
     while (mmuSerial.available()) {
-        rx_buffer[i++] = mmuSerial.read();
-        rx_buffer[i] = '\0';
-
-        if (i == sizeof(rx_buffer) - 1) {
+        if (i == sizeof(rx_buffer) - 2) {
```

Comment:
Since we're doing a safety check here, it would make more sense to use `>=`.

Reply:
indeed.